README.md

```diff
@@ -173,6 +183,73 @@ Kind: *Trusted Indication*.
 
 Errors:
 
+### `QueryPallet`
+
+Query the existence of a particular pallet type.
+
+- `name: Vec<u8>`: The name of the pallet to query.
+- `query_id`: The value to make the returned message identifiable with this query.
+
+Returns: A honestly populated `PalletInfo` to the origin.
+
+Safety: No concerns.
+
+Kind: *Instruction*
+
+Errors:
+
+### `PalletInfo`
+
+Provide information on the existence of a particular pallet type.
+
+- `query_id` The identifier of the query which caused this message to be sent.
+- `pallets: Vec<(u32, (u32, u32, u32))>`: The index/version of each of the pallets whose name matches the name in the according `QueryPallet` message.
```

Comment:
How do we differentiate multiple instances? Add another `Vec<u8>` which can be pallet prefix?

Reply:
the index (first item in tuple) will differentiate between instances.